core/src/main/java/org/apache/calcite/rel/logical/LogicalFilter.java

```diff
@@ -135,9 +133,7 @@ public static LogicalFilter create(final RelNode input, RexNode condition,
     final RelMetadataQuery mq = cluster.getMetadataQuery();
     final RelTraitSet traitSet = cluster.traitSetOf(Convention.NONE)
         .replaceIfs(RelCollationTraitDef.INSTANCE,
-            () -> RelMdCollation.filter(mq, input))
-        .replaceIf(RelDistributionTraitDef.INSTANCE,
-            () -> RelMdDistribution.filter(mq, input));
+            () -> RelMdCollation.filter(mq, input));
```

Comment:
I don't know how this change is related.

Reply:
If we don't do this. 
testEnumerableFilterRule will throw an exception:
```
There are not enough rules to produce a node with desired properties: convention=ENUMERABLE, dist=any.
Missing conversion is LogicalFilter[convention: NONE -> ENUMERABLE]
There is 1 empty subset: rel#39:RelSubset#1.ENUMERABLE.broadcast, the relevant part of the original plan is as follows
14:LogicalFilter(condition=[NOT(<= SOME($5, {
LogicalProject(COMM=[$6])
  LogicalTableScan(table=[[CATALOG, SALES, EMP]])
}))])
  8:LogicalTableScan(subset=[rel#13:RelSubset#0.NONE.any], table=[[CATALOG, SALES, EMP]])
```
The mismatch of Trait information leads to the invalidity of the transformed plan.

Reply:
I'm not sure if this is a correct modification because no other test cases cover this point.

Reply:
Create CALCITE-6899 to log this issue and use a dynamic table without RelDistributionTraitDef to ensure the test passes.

Reply:
I try to remove `p.addRelTraitDef(RelDistributionTraitDef.INSTANCE)`, error can not throw.

Reply:
@xiedeyantu We can discuss this issue in CAICITE-6899